apps/src/javalab/JavalabEditor.jsx

```diff
@@ -360,6 +362,46 @@ class JavalabEditor extends React.Component {
     });
   }
 
+  // This moves the active tab to the left in the tab menu
+  moveTabLeft() {
+    const {activeTabKey} = this.props;
+    const {orderedTabKeys, setOrderedTabKeys} = this.props;
+    let newTabs = [...orderedTabKeys];
+    let index = newTabs.indexOf(activeTabKey);
+    if (index > 0) {
+      let fileToMoveLeft = newTabs[index];
+      newTabs[index] = newTabs[index - 1];
+      newTabs[index - 1] = fileToMoveLeft;
+      setOrderedTabKeys(newTabs);
+    }
+
+    // closes the tab menu if it is open
+    this.setState({
+      showMenu: false,
+      contextTarget: null
+    });
+  }
+
+  // This moves the active tab to the right in the tab menu
+  moveTabRight() {
+    const {activeTabKey} = this.props;
+    const {orderedTabKeys, setOrderedTabKeys} = this.props;
+    let newTabs = [...orderedTabKeys];
+    let index = newTabs.indexOf(activeTabKey);
+    if (index < newTabs.length - 1) {
```

Comment:
can you make a helper method to consolidate the logic here? For example, something like 
```
swapTabs(index1, index2) {
  // handle swapping tabs and closing the tab menu
}
```
You can probably consolidate everything by checking that both indices are not out of bounds (i.e. just check `>= 0` and `< newTabs.length` for each index)

Reply:
Thanks so much! I created a helper function to swap tabs, set key, and close tab. I didn't end up consolidating everything because to move tab right, the index has to be >0, not >=0, ...

Reply:
Just to follow up here, I think you could still consolidate the logic as Molly suggested. Since you're passing both indices, you just need to make sure both are valid and it should work for both cases. Ex. for moving left on the first tab, you'd end up passing in index 0 and index -1 (invalid).

Reply:
Oh - gotcha! Thanks @sanchitmalhotra126! and Thanks @molly-moen for the original suggestion! Will do.